STYLEGUIDE.md

```diff
@@ -2,6 +2,33 @@
 
 For matters which aren't mentioned in this file, refer to [PEP8](https://peps.python.org/pep-0008/).
 
+## Linting tools
+To check linting we are using: [flake8](https://flake8.pycqa.org/en/latest/), [black](https://black.readthedocs.io/en/stable/), [isort](https://pycqa.github.io/isort/), and [pre-commit](https://pre-commit.com/).
+Corresponding configurations are added to the [backend](https://github.com/YangCatalog/backend), [sdo_analysis](https://github.com/YangCatalog/sdo_analysis), and [yang-validator-extractor](https://github.com/YangCatalog/yang-validator-extractor) repositories for now.
+
+**Useful notes for flake8** (Other configuration details for every tool can be found in documentations):
+
+- If there's a flake8 error for the exact line, but this line needs to stay as it is - just simply add ```# noqa``` comment at the end of this line, and it will disable flake8 for this line. It's also possible to indicate some specific errors in this comment, for example, ```# noqa: F401, F405``` will only disable ```F401``` and ```F405``` errors for this line but will still check this line for other errors.
+- If you are using [per-file-ignores](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-per-file-ignores) and run ```flake8 dir_name --config=./dir_name/.flake8```  from the ```development``` directory, then these ```per-file-ignores``` will not work , but running it from the ```dir_name``` directory will enable per-file-ignores correctly. So, if you want to run flake8 from the development directory to see all the navigation urls but don't want to see the navigation urls to these ignored files, then you have to add ```*/``` at the beginning of every path in ```per-file-ignores```, so ```actual/path/to/ignore.py``` will become ```*/actual/path/to/ignore.py``` (don't forget to change it back after).
+
+**There is a guide to enable tools locally** (The ```backend``` repository is used in this example, but all the steps can be used in every other repository listed above):
+
+**Required steps:**<br>
+Go to the ```backend``` directory and run the following commands
+```
+pip install pre-commit
+pre-commit install
+pre-commit autoupdate
+```
+It will enable the pre-commit hook, so all your future commits will automatically be reformatted by isort and black, and also they will be checked by flake8. If you don't do it, then all the errors found by any of these tools will break your GitHub Actions and you'll be forced to fix them anyway.
+
+**Recommended steps:**<br>
+It's convenient to be able to run any of these tools locally, so I recommend to install all other dependencies as well: ```pip install flake8 flake8-black flake8-isort flake8-absolute-import flake8-commas flake8-comprehensions flake8-docstrings flake8-multiline-containers flake8-print flake8-quotes pep8-naming```, and now you'll be able to run all the tools locally:
```

Comment:
These also can be installed by running `pip install -r lint_requirements.txt`, maybe we should mention it too.